src/components/_global/BalForm/useForm.ts

```diff
@@ -0,0 +1,134 @@
+import { Rules } from '@/types';
+import { onBeforeMount, reactive, toRefs, ToRefs, Ref } from 'vue';
+
+type UseFormRequest<DefaultValues> = {
+  name: string;
+  defaultValues?: Partial<
+    { [Property in keyof DefaultValues]: Ref<any> | any }
+  >;
+};
+
+export type FormState = {
+  name: string;
+  values: Record<string, any>;
+  errors: Record<string, string[]>;
+  rules: Record<string, Rules | undefined>;
+  defaultValues: Record<string, any>;
+};
+
+export type FormInstance = ToRefs<FormState> & {
```

Comment:
What is this ToRefs with an ampersand? I haven't seen it before and can't find any docs on it. Is it making `FormInstance` extend `FormState` with the same parameters but they are all references to the same types specified in FormState?